src/index.jsx

```diff
@@ -65,14 +73,6 @@ const injectBarInDOM = data => {
 }
 
 const renderBar = (barNode, options) => {
-  // import React related modules on init only
-  const React = require('react')
-  const { render } = require('react-dom')
-  const { connect, Provider } = require('react-redux')
-  const I18n = require('cozy-ui/react/I18n').default
-  const Bar = require('components/Bar').default
-  const CozyProvider = require('cozy-client').CozyProvider
```

Comment:
By removing those lines, we'll not be able to build the old style bar (by old style bar I mean... the bar that injects react by itself)

Reply:
Why ?

Reply:
Because now, if you load the script bar before your app script, you'll have an error since React will not be exposed yet, no ? 

cf https://github.com/cozy/cozy-passwords/blob/master/src/targets/browser/index.ejs / https://github.com/cozy/cozy-sandbox/blob/master/src/targets/browser/index.ejs /

Reply:
For the full bundle, it will use the React from the bundle.
For the bundle without React, the cozy-bar must be included after the app.

I think it will simplify our mental model, either you use the full bundle and the bar uses its own React, or it does not include it and it uses the React from the app.

Reply:
> For the bundle without React, the cozy-bar must be included after the app.

This is my point. It's a BC for a lot of Cozy app's. I don't know if we want to upgrade all of our old apps for that (mesinfos and other apps like that)

Reply:
Older apps will not use the bundle without React, this is opt-in.

Reply:
OK, I had a misunderstanding on my side between bundled and bundle without react, sorry for that. 

I understand the fact we "external" React & ReactDom, I understand that if we inject React in window.React, this window.React will be used but: 

- Is it because any external module will be replaced by window.MyModule by webpack automatically? So we we write `import React from 'react'` , it'll be transformed to import React from window.react ? 

- It seems that we don't use globalReact anymore, so how the bundled bar require React & ReactDom (or ATM called 'my-react')?

Reply:
>  Is it because any external module will be replaced by window.MyModule by webpack automatically? So we we write import React from 'react' , it'll be transformed to import React from window.react ?

The end result is yes, but I think to achieve this, webpack simply creates a "fake" React module that exports window.React. But it has the same end effect as what you wrote.

>  It seems that we don't use globalReact anymore, so how the bundled bar require React & ReactDom (or ATM called 'my-react')?

It requires it through a normal import. It is not there for the moment in the PR, this might be why it is not clear, but there will be two builds, one with `webpackConfig.externals` configured to externalize React and ReactDOM and the other without this `externals` config.